sdv/data_processing/data_processor.py

```diff
@@ -114,6 +115,8 @@ def __init__(
         locales=['en_US'],
     ):
         self.metadata = metadata
+        if isinstance(metadata, Metadata):
+            self.metadata = metadata._convert_to_single_table()
```

Comment:
this situation shouldn't ever occur since we directly pass the metadata to the data processor